WalletWasabi.Fluent/App.axaml.cs

```diff
@@ -79,20 +77,62 @@ public override void OnFrameworkInitializationCompleted()
 		base.OnFrameworkInitializationCompleted();
 	}
 
+	private Window CreateMainWindow()
+	{
+		var result = new MainWindow
+		{
+			DataContext = MainViewModel.Instance
+		};
+
+		Observable.FromEventPattern(result, nameof(result.Closed))
```

Comment:
I might be wrong but shouldn't this be disposed?

Every time when `IMainWindowService.Show()` is called, it creates a new `MainWindow` and a subscription happens on its `Closed` event.

In `IMainWindowService.Close()` the `MainWindow` is set to `null` so GC could collect it theoretically, but due to this subscription... Won't GC skip it?

Reply:
@soosr is right. The subscription to the Window.Closed event is never disposed. 

Maybe you could do something like this:

```
subscription?.Dispose();
var window = new MainWindow
{
    DataContext = MainViewModel.Instance;
}

var subscription = Observable.FromEventPattern(result, nameof(window.Closed)) { ... }
```

This way, if you ever call `CreateMainWindow()` multiple times, the previous subscription will be disposed correctly.